libbeat/docs/command-reference.asciidoc

```diff
@@ -0,0 +1,583 @@
+
+//REVIEWERS: It seems odd to me that we usually expect comma-separated lists, but here we expect a space-separated list.
+
+*`enable MODULE_LIST`*::
+Enables the modules specified in the space-separated list.
+
+*`list`*::
+Lists the modules that are currently enabled and disabled.
+
+
+*FLAGS*
+
+*`-h, --help`*::
+Shows help for the `export` command.
+
+
+{global-flags}
+
+*EXAMPLES*
+
+ifeval::["{beatname_lc}"=="filebeat"]
+
+["source","sh",subs="attributes"]
+-----
+{beatname_lc} modules list
+{beatname_lc} modules enable apache2 auditd mysql
+-----
+
+endif::[]
+
+ifeval::["{beatname_lc}"=="metricbeat"]
+
+["source","sh",subs="attributes"]
+-----
+{beatname_lc} modules list
+{beatname_lc} modules enable apache nginx system
+-----
+
+
+endif::[]
+
+endif::[]
+
+
+[[run-command]]
+==== `run` command
+
+{run-command-short-desc}.
+
+*SYNOPSIS*
+
+["source","sh",subs="attributes"]
+-----
+{beatname_lc} run [FLAGS]
+-----
+
+
+*FLAGS*
+
+ifeval::["{beatname_lc}"=="packetbeat"]
+
+*`-I, --I FILE`*::
+Reads packet data from the specified file instead of reading packets from the
+network. This option is useful only for testing {beatname_uc}.
++
+["source","sh",subs="attributes"]
+-----
+{beatname_lc} run -I ~/pcaps/network_traffic.pcap
+-----
+
+endif::[]
+
+*`-N, --N`*::
+Disables the publishing of events to the defined output. This option is useful
+only for testing {beatname_uc}.
+
+ifeval::["{beatname_lc}"=="packetbeat"]
+
+*`-O, --O`*::
+Read packets one by one by pressing _Enter_ after each. This option is useful
+only for testing {beatname_uc}.
+
+endif::[]
+
+*`--cpuprofile FILE`*::
+Writes CPU profile data to the specified file. This option is useful for
+troubleshooting {beatname_uc}.
+
+ifeval::["{beatname_lc}"=="packetbeat"]
+
+*`-devices`*::
+Prints the list of devices that are available for sniffing and then extis.
+
+endif::[]
+
+ifeval::["{beatname_lc}"=="packetbeat"]
+
+*`-dump FILE`*::
+Writes all captured packets to the specified file. This option is useful for
+troubleshooting {beatname_uc}.
+
+endif::[]
+
+*`-h, --help`*::
+Shows help for the `run` command.
+
+*`--httpprof [HOST]:PORT`*::
+Starts an http server for profiling. This option is useful for troubleshooting
+and profiling {beatname_uc}.
+
+ifeval::["{beatname_lc}"=="packetbeat"]
+
+*`-l N`*::
+Reads the pcap file `N` number of times. The default is 1. Use this option in
+combination with the `-I` option. For an infinite loop, use _0_. The `-l`
+option is useful only for testing {beatname_uc}.
+
+endif::[]
+
+*`--memprofile FILE`*::
+Writes memory profile data to the specified output file. This option is useful
+for troubleshooting {beatname_uc}.
+
+ifeval::["{beatname_lc}"=="filebeat"]
+
+//REVIEWERS: I've written this topic to be consitent to what I see when I run -help, but I'm wondering why we aren't also showing the options that begin with a dash. For example, in the modules doc we show: -modules=nginx,mysql,system. Is --modules preferred?
```

Comment:
For now, I'll keep the docs consistent with what we show in -h and change them when we update the help. I think it's probably good to keep the -options around for now because it might help existing users to see familiar commands. I do think, though, that we should consider using the --options in the getting started if those are preferred.